docs/platform/porting/system.md

```diff
@@ -2540,3 +2540,356 @@ Example:
   $ sleep 123 &
   $ livedumper -f sleep.core -m $?
 ```
+
+### stability-monitor
+
+**stability-monitor** purpose it to monitor usage of system resources with low overhead. The resources tracked are - maximum and average usage of CPU, memory, I/O. Additionally, number of open file descriptors is tracked. Resources are consumed by various processes, which can be classified according to their type as application or service. Applications have type (native, webapp, csharp) and state (foreground, background). The process parameters are distinguished in configuration and various parameters can be assigned to different services and applications. When an excessive use of resources (abnormality) is detected, appropriate D-Bus signal is sent and (optional) abnormality report is generated.
+
+#### Kernel module
+
+**stability-monitor** relies on kernel module, which collects information about processes state in the system in most resource efficient way. This raw data is exposed via `/proc/tsm` interface.
+
+The following resource statistics are collected:
+-  CPU - system time + user time [cpu_usage = d(system_time + user_time) / dt]
+-  Memory - RSS
+-  I/O - bytes read + bytes written
+-  FDs - number of open file descriptors
+
+#### Service
+
+Purpose of **stability-monitor** service is to:
+-  Periodically monitor process state (based on raw data provided by kernel module)
+-  Gather additional process specific data from running system (eg. app type & state from AUL and pkgmgr)
+-  Read and apply provided configuration to the system (apply configuration defined limits)
+-  Generate D-Bus signal (and abnormality report, if configured)
+
+Service monitors parameters as provided by "data source" modules (which are processing raw data provided by kernel module).
+
+#### Configuration
+
+The configuration uses a hierarchical layout which consists of a set of JSON files. The primary file is `/usr/lib/stability-monitor/default.conf`. Additional files can be placed in `/etc/stability-monitor.d` and are read in alphabetical order (such that `Z.conf` will override `A.conf`). Configuration files are merged together (not replaced).
+
+In the resulting JSON object, the top-level entries have keys corresponding to programs, for example:
+-  resourced
+-  org.tizen.calendar
+
+Program name should be application ID for applications and executable basename (from `/proc/<pid>/exe`) for other system processes.
```

Comment:
Can you please rephrase, itis little confusing. 
Program name must bee application ID for applications and executable basename (from `/proc/<pid>/exe`) for other system processes.